### ISSUE TITLE
Analyze each etcd test runtime

### ISSUE THREAD

**Issue by chaochn47** [etcd-io/etcd#16080]
## Background

`go test -v -json <TestCase>`  + [gotestsum](https://pkg.go.dev/gotest.tools/gotestsum#readme-junit-xml-output) will write a test report post run.

This report will contain each test suite (for example, robustness test), each test case (including subtest case) duration, success or not, skipped or not, etc. 

This test report has been uploaded in the integration test suite and used in [measure-test-flakiness workflow](https://github.com/etcd-io/etcd/actions/workflows/measure-test-flakiness.yaml)  https://github.com/etcd-io/etcd/blob/b46dc0a1e4866afacc24f3fc438703ca3d2f97f4/.github/workflows/tests.yaml#L55-L58

To get a sense of what the produced Junit xml output looks like, download the artifacts in the bottom right corner of the page in https://github.com/etcd-io/etcd/actions/runs/5268307246. 

## What to be added

- [ ] Expand uploading test artifacts generated in the integration test suite to robustness test and e2e test suite CI (including the equivalent tests running on arm64). 
- [ ] Analyze the test case runtime distribution of a test suite. 

### Why is this needed?

To help analyze each test case runtime and eventually reduce the presubmit tests runtime. For example https://github.com/etcd-io/etcd/issues/15086. Presubmit tests over 30mins is not good for developer experience and development velocity.

**Comment by chaochn47** [comment]
Thanks! @sbdtu5498 Assigned to you.

**Comment by siddharth-sable** [comment]
is anyone working on this?

**Comment by jmhbnz** [comment]
I don't believe so @siddharth-sable - we certainly welcome new contributors if you're interested in picking it up?

**Comment by siddharth-sable** [comment]
Yes, I am interested in picking it up

**Comment by serathius** [comment]
We get similar benefits from testgrid
